pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/impl/PersistentTopicsBase.java

```diff
@@ -4414,24 +4414,20 @@ protected void handleTopicPolicyException(String methodName, Throwable thr, Asyn
     }
 
     protected void internalTruncateNonPartitionedTopic(AsyncResponse asyncResponse, boolean authoritative) {
-        Topic topic;
-        try {
-            validateAdminAccessForTenant(topicName.getTenant());
-            validateTopicOwnership(topicName, authoritative);
-            topic = getTopicReference(topicName);
-        } catch (Exception e) {
-            log.error("[{}] Failed to truncate topic {}", clientAppId(), topicName, e);
-            resumeAsyncResponseExceptionally(asyncResponse, e);
-            return;
-        }
-        CompletableFuture<Void> future = topic.truncate();
-        future.thenAccept(a -> {
-            asyncResponse.resume(new RestException(Response.Status.NO_CONTENT.getStatusCode(),
-                    Response.Status.NO_CONTENT.getReasonPhrase()));
-        }).exceptionally(e -> {
-            asyncResponse.resume(e);
-            return null;
-        });
+        CompletableFuture.supplyAsync(()->{
+                    validateAdminAccessForTenant(topicName.getTenant());
+                    return null;
+                })
+                .thenCompose(__ -> validateTopicOwnershipAsync(topicName, authoritative))
+                .thenCompose(__ -> getTopicReferenceAsync(topicName))
+                .thenCompose(topic -> topic.truncate())
+                .thenAccept(a -> {
+                    asyncResponse.resume(new RestException(Response.Status.NO_CONTENT.getStatusCode(),
+                            Response.Status.NO_CONTENT.getReasonPhrase()));
+                }).exceptionally(e -> {
```

Comment:
1. Add an error log here.
2. We need to unwrap exception of type CompletionException and ExecutionException by call `ex.getCause()`